core/lib/src/doc/has_schema.rs

```diff
@@ -0,0 +1,318 @@
+pub enum SchemaKind {
+    Null,
+    Map,
+    List,
+    String,
+    Num,
+    Int,
+    Bool,
+    Set,
+}
+
+pub struct TypeSchema<T> {
```

Comment:
This doesn't make much sense to me.￼ `name` seems to be a string serialization of the kind enum, so why have it at all? Or alternatively, why not have the trait simply have a const string?￼￼

I think we need to start thinking more seriously about how schema is going to be represented.￼ Otherwise, we will be bouncing back-and-forth between ideas without justification. A good approach here is the take a reasonably sized application and imagine the exact documentation we would like to render and work backwards from there. We also have the benefit of existing schema like JSON schema and so on. ￼